### PR TITLE
Adds functionality of Listing notes

### PR DESCRIPTION
**Pull request by JessicaLim8**
### New functions
Allows for lists to be displayed on console via `note list <filename>` or using the `-l` or `--list` flag with existing commands. Closes issue #15 

#### Note List command

- Will list all the elements within the called filename
- If the list doesn't exist, it will be made
- If no list is included, the default list will be displayed

#### Note List features
- Will display the title, coloured by property
- Will display each element in the appropriate colour
- Elements are displayed with either checkmarks/crosses if they are of checkbox type, or points, if they are dot-jot notes (according to the filename's properties
- If the element is empty, it will not be displayed

#### Listing flags

- If `-l` or `--list` is flagged in any command, after the command is completed, the altered list is displayed
- Default of `-l` is false (will not list elements unless flag is present)
- If flag is pulled with conjunction with `note list` i. e. `note list -f <listname> -l`, it will only list once

### CODE REVIEW

**Review by JonathanGin52 — CHANGES_REQUESTED** [review]
Review hound's comments

**Comment by JessicaLim8** [comment]
Made the appropriate changes to address formatting issues

File: commands/list.js [hunk 1]
```diff
@@ -1,4 +1,28 @@
...
+  let checkbox;
+  console.log(title[titleColour] + '\n');
+
+  if (entries.length === 0) {
+    error('There are no entries in this list', true, 0);
+  }
+  for (let i = 0; i < entries.length; i++) {
+    if (type === 'Dot-jots') {
+      checkbox = ' ●';
+    } else if (entries[i].checked === true) {
```
Comment (JonathanGin52) [inline commands/list.js]: Remove redundant `=== true`

File: commands/list.js [hunk 2]
```diff
@@ -1,4 +1,28 @@
+const dir = `${require('os').homedir()}/.terminote/`;
+const color = require('colors');
```
Comment (JonathanGin52) [inline commands/list.js:1]: Redundant import

File: commands/list.js [hunk 3]
```diff
@@ -1,4 +1,28 @@
...
+const error = require('../util/error');
+
 module.exports = (args) => {
-  console.log('Here are your notes!');
+  const file = require(dir + args.f);
+  const { title, titleColour, colour, type, entries } = file;
```
Comment (JonathanGin52) [inline commands/list.js]: combine line 6 and line 7 since file isn't being used anywhere else

File: commands/list.js [hunk 4]
```diff
@@ -1,4 +1,28 @@
...
+  let checkbox;
+  console.log(title[titleColour] + '\n');
+
+  if (entries.length === 0) {
```
Comment (JonathanGin52) [inline commands/list.js:9]: Could be replaced by `if (!entries.length)` because 0 is falsy. The current syntax can be kept for clarity

**Review by JonathanGin52 — APPROVED** [review]
Looks good

**Pull request merged**